src/Modules/Authontication/Components/Context/AuthContext.tsx

```diff
@@ -0,0 +1,29 @@
+
+import { jwtDecode } from "jwt-decode";
+import { PropsWithChildren, useEffect, useState } from "react";
+import { createContext } from "react";
+
+export let AuthContext = createContext(null);
+
+export default function AuthContextProvider(props:   
+ PropsWithChildren) {
+  let [loginData, setLoginData] = useState(null);
+
+  let saveLoginData = () => {
+    const encodedToken = localStorage.getItem('token');
+    if (encodedToken) {
+      const decodedToken = jwtDecode(encodedToken);
```

Comment:
I suggest calling the currentUser api directly after login and then store the data in the localStorage, and get the full data from the backend as it contains the user's avatar and this will enable us to remove the jwt-decode package.